daemon/cmd/fqdn.go

```diff
@@ -92,29 +93,34 @@ func (d *Daemon) updateSelectors(ctx context.Context, selectors map[policyApi.FQ
 		updateResult |= res
 	}
 
+	var policyMapsDone channels.DoneChan
+
 	// UpdatePolicyMaps consumes notifyWG, and returns its own WaitGroup
 	// that is Done() when all endpoints have pushed their incremental changes
 	// down in to their bpf PolicyMap.
 	if updateResult&policy.UpdateResultUpdatePolicyMaps > 0 {
 		logger.Debug("FQDN selector update requires UpdatePolicyMaps.")
-		wg = d.endpointManager.UpdatePolicyMaps(ctx, notifyWg)
+		policyMapsDone = d.endpointManager.UpdatePolicyMaps(ctx, notifyWg)
 	} else {
-		wg = &sync.WaitGroup{}
+		policyMapsDone = channels.ClosedDoneChan
+	}
+
+	if updateResult&policy.UpdateResultIdentitiesNeeded == 0 || ipcacheRevision == 0 {
+		return policyMapsDone
 	}
 
 	// If any of the selectors indicated they're missing identities,
 	// we also need to wait for a full ipcache round.
-	if updateResult&policy.UpdateResultIdentitiesNeeded > 0 && ipcacheRevision > 0 {
-		wg.Add(1)
-		go func() {
-			logger.Debug("FQDN selector update requires IPCache completion.")
-			d.ipcache.WaitForRevision(ipcacheRevision)
-			wg.Done()
-		}()
-	}
+
+	ipCacheDone := make(chan struct{})
+	go func() {
+		logger.Debug("FQDN selector update requires IPCache completion.")
+		d.ipcache.WaitForRevision(ipcacheRevision)
+		close(ipCacheDone)
+	}()
 
 	// This releases the nameManager lock; at this point, it is safe for another fqdn update to proceed
-	return
+	return channels.Merge(policyMapsDone, ipCacheDone)
```

Comment:
Goroutines are spawned here as per my previous comment. I'm trying to understand how this fits into the part of the commit msg which states:

> Refactor UpdatePolicyMaps to leverage the triggering behaviour and to
avoid spawning a goroutine per endpoint.

Reply:
I see that the comment was specific to `UpdatePolicyMaps`. However, I think my general point still stands: it's not clear what the specific improvement over goroutines are given the implications of `channels.Merge()`. How much have we reduced goroutines, or are we still using the same amount?